File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        res = ""
        res_len = 0

        for i in range(n):
            low = i
            high = i

            while low >= 0 and high < n and s[low] == s[high]:
                if high - low + 1 > res_len:
                    res_len = high - low + 1
                    res = s[low:high+1]

                low -= 1
                high += 1
            
            low = i
            high = i+1

            while low >= 0 and high < n and s[low] == s[high]:
                if high - low + 1 > res_len:
                    res_len = high - low + 1
                    res = s[low:high+1]

                low -= 1
                high += 1
        
        return res

                

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.size();
        if (n == 1) {
            return s;
        }
        string result(1, s[0]);
        for (int i = 1; i < n; i++) {
            int low = i;
            int high = i;

            while (low >= 0 && high < n && s[low] == s[high]) {
                low--;
                high++;
            }
            string check = s.substr(low + 1, high - low - 1);
            if (check.size() > result.size()) {
                result = check;
            }

            low = i - 1;
            high = i;

            while (low >= 0 && high < n && s[low] == s[high]) {
                low--;
                high++;
            }
            check = s.substr(low + 1, high - low - 1);
            if (check.size() > result.size()) {
                result = check;
            }
        }
        return result;
    }
};